File: frontend/src/types/website.ts
export interface Website {
  id: string;
  name: string;
  url: string;
  status: string;
  status_code: number | null;
  response_time: number | null;
  ssl_expiry: string | null;
  last_check: string | null;
  created_at: string;
  updated_at: string;
}

export interface WebsiteCreate {
  name: string;
  url: string;
}

File: backend/app/schemas/website.py
from uuid import UUID
from datetime import datetime

from pydantic import BaseModel, ConfigDict


class WebsiteCreate(BaseModel):
    name: str
    url: str


class WebsiteRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: UUID
    name: str
    url: str

    status: str
    status_code: int | None = None
    response_time: float | None = None

    ssl_expiry: datetime | None = None
    last_check: datetime | None = None

    created_at: datetime
    updated_at: datetime